src/waveform/renderers/waveformmark.h

```diff
@@ -23,31 +31,27 @@ class WaveformMark {
     WaveformMark& operator=(const WaveformMark&) = delete;
 
     // Enable swapping
-    void swap(WaveformMark& that) {
-        std::swap(m_pPointCos, that.m_pPointCos);
-        std::swap(m_properties, that.m_properties);
-        std::swap(m_iHotCue, that.m_iHotCue);
-        std::swap(m_image, that.m_image);
-    }
-    friend void swap(WaveformMark& lhs, WaveformMark& rhs) {
-        lhs.swap(rhs);
-    }
-
-    void reset(int hotCue = kDefaultHotCue);
-
-    void setup(const QString& group, const QDomNode& node,
-               const SkinContext& context,
-               const WaveformSignalColors& signalColors);
-
-    std::unique_ptr<ControlProxy> m_pPointCos;
+    // Swapping is done in the destructor
+    
 
     const WaveformMarkProperties& getProperties() const { return m_properties; };
     void setProperties(const WaveformMarkProperties& properties) { m_properties = properties; };
 
     int getHotCue() const { return m_iHotCue; };
     void setHotCue(int hotCue) { m_iHotCue = hotCue; };
 
+    //The m_pPointCos related function
+    bool is_m_pPointCos_Null() const{ return m_pPointCos == nullptr; }
```

Comment:
Please tell me if it is not needed, then what should I do at the positions, where it was checking that 

`if(pMark->m_pPointCos)`
I thought that this is the only meaning behind it, so I had no choice, you are right that we should not reveal the private member of the class, its a bad practice of OOP, but if anything I can do please tell